source/slang/slang-ir-byte-address-legalize.cpp

```diff
@@ -92,6 +92,11 @@ struct ByteAddressBufferLegalizationContext
         // Get the equivalent structured buffer for the buffer.
         if( auto structuredBuffer = getEquivalentStructuredBuffer(elementType, buffer) )
         {
+            if (buffer != structuredBuffer)
+            {
+                buffer->replaceUsesWith(structuredBuffer);
+                buffer->removeAndDeallocate();
+            }
```

Comment:
I am trying to understand why this is needed.
Can you add a comment for this block so that other people may not accidently remove it in a long term?